### PR TITLE
Refactor for dryer code

### PR DESCRIPTION
**Pull request by jaharaclark**
## Pull Request

<br>

### Description

* Is this a feature or a fix?
This is a fix to make our code dryer after completion.

* Where should the reviewer start?
There isn't anywhere specific to start, it's just an overall polish. 

* What functionalities do these changes effect? Why were these changes made? What was the motivation behind these changes?
This was designed to make our code cleaner with more developer empathy.

* What outside features or research did you use to implement this fix?

* How should this be tested?
The functionality should be the exact same.

<br>

***

#### Please review considerations below:

- [x] Follows Turing Style Guidelines
- [x] Changes generate no new warnings or errors
- [x] Checked code for and corrected any spelling errors
- [x] README has been updated(if applicable)

### CODE REVIEW

**Comment by aemiers** [comment]
Thanks for your awesome work!

File: src/main.js [hunk 1]
```diff
@@ -167,40 +166,23 @@ function searchBarSearch () {
...
-//   }
-// }
+function showStarredIdeas() {
+  changeStarredIdeasButtonText();
+  retrieveFromStorage();
+  filterIdeas();
+}
+
+function filterIdeas() {
+  var allCards = document.querySelectorAll('.idea-card');
```
Comment (joel-oe-lacey) [inline src/main.js:178]: Nice use of querySelectorAll!

File: src/main.js [hunk 2]
```diff
@@ -167,40 +166,23 @@ function searchBarSearch () {
...
+  for (var i = 0; i < allCards.length; i++) {
+    if (!allSavedIdeas[i].star) {
+      allCards[i].classList.toggle('hidden');
+    }
+  }
+}
+
+function changeStarredIdeasButtonText() {
+  if (showStarredIdeasButton.innerHTML === 'Show Starred Ideas') {
+    showStarredIdeasButton.innerHTML = 'Show All Ideas';
```
Comment (joel-oe-lacey) [inline src/main.js:188]: This is a one-way journey right now, doesn't swap back to "Show Starred Ideas"

File: src/idea.js [hunk 3]
```diff
@@ -10,18 +10,18 @@ class Idea {
...
+//       }
+//     }
+//     localStorage.setItem("saved-ideas", JSON.stringify(savedIdeas))
+//   }
+//
+//   updateIdea() {
+//     this.star = !this.star;
+//     // this.star ? this.src = "images/star-active.svg" : this.src = "images/star.svg"
+//   }
+// }
```
Comment (joel-oe-lacey) [inline src/idea.js]: Be careful with temporary commenting, this currently breaks the syntax of this class.

**Review by joel-oe-lacey — APPROVED** [review]
Lots of good functionality here! Some really neat and clever consolidation, I like the use of .closest. 

Noticed a few small things fiddling with the site that aren't in this change. 

You can currently add a blank idea, perhaps research if there’s a way to disable a button by default. 

Your comment button uses the same class as the delete button, which has some side effects. 

In terms of styling:
- I like the hover animations.
- Good job handling text overflow on the boxes, though you may want to wrap body text around so it can take up more than one line. 
- Neat experimenting with grid. 
- The .idea-cards-section as it's currently set up is a bit too wide.
- Pointer cursors on the buttons, sweet.  

You might already be on top of some of those! All in all good job working through the struggles, love the enthusiasm on the readme.

**Pull request merged**